conv_model_fwd.cpp

```diff
@@ -371,7 +396,7 @@ int conv_benchmark(int argc, char** argv) {
             if (input_padding_copy)
               gemm_param.b.primary = LIBXSMM_ACCESS_RAW(5, sizeof(DType), scratch_input_libxsmm, i_n, i_c, i_h * stride_h + i_r, i_w * stride_w + i_s, 0, Cb, ifhp_physically_padded, ifwp_physically_padded, bc);
             else
-              gemm_param.b.primary = LIBXSMM_ACCESS_RAW(5, sizeof(DType), input_libxsmm, i_n, i_c, i_h * stride_h + i_r, i_w * stride_w + i_s, 0, Cb, ifhp, ifwp, bc);
+              gemm_param.b.primary = LIBXSMM_ACCESS_RAW(5, sizeof(DType), input_libxsmm, i_n, i_c, pad_h_in - pad_h + i_h * stride_h + i_r, pad_w_in - pad_w + i_w * stride_w + i_s, 0, Cb, ifhp, ifwp, bc);
```

Comment:
@egeor the new expression with `pad_h_in - pad_h` and such looks strange too me but this is the one which works.
Let me know if you think this is obviously wrong.

Reply:
Wouldn't it work if you were offseting the original pointer, i.e. what we do for output:
https://github.com/libxsmm/parlooper/blob/8e1b136218ed071490ad221617ee75e86d7992ef/conv_model_fwd.cpp#L131

Reply:
I will not get the `pad_h_in - pad_h` only from the offset (or, if I shift by the physical vs logical input padding, it would not be the same "offset" as for the output). If I just offset by `pad_h_in`, `pad_w_in` then the expression in the changed lines anyway will be looking different than in other accesses to input.
So I don't see how this will make things looking better.

Reply:
Yes, I agree. My point was merely that introducing `input_libxsmm_off` with offset defined through  `pad_h_in - pad_h`, `pad_w_in - pad_w` and using it in some but not all places (as we have a differently offset-ed access to input in avoid_rim_fmas = 1 code path) doesn't look better to me than the current code lines.
So unless you have a strong preference, I'd keep the commit changes as they are now.

Reply:
OK I see, now. I don't have strong feelings so we can keep as is.